v2/dtos/reading.go

```diff
@@ -43,23 +49,50 @@ type BinaryReading struct {
 // https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.x#/SimpleReading
 type SimpleReading struct {
 	Value         string `json:"value" validate:"required"`
-	ValueType     string `json:"valueType" validate:"required"`
 	FloatEncoding string `json:"floatEncoding,omitempty"`
 }
 
+// Validate satisfies the Validator interface
+func (b BaseReading) Validate() error {
+	// check if the inner BinaryReading struct is empty
+	if !reflect.DeepEqual(b.BinaryReading, BinaryReading{}) {
```

Comment:
Now, you should be able to check `ValueType == ValueTypeBinary`  rather than do this Deep Equal reflection check. If not Binary, assume `SimpleReading`. i.e. no need to final `else` on line 75

Reply:
Updated to check inner struct by `ValueType` instead of reflect, thanks.